src/Microsoft.AspNetCore.Sockets.Client.Http/ServerSentEventsTransport.cs

```diff
@@ -68,17 +69,24 @@ public Task StartAsync(Uri url, IDuplexPipe application, TransferMode requestedT
                 return t;
             }).Unwrap();
 
-            return Task.CompletedTask;
+            return startTcs.Task;
         }
 
-        private async Task OpenConnection(IDuplexPipe application, Uri url, CancellationToken cancellationToken)
+        private async Task OpenConnection(IDuplexPipe application, Uri url, TaskCompletionSource<object> tcs, CancellationToken cancellationToken)
         {
             Log.StartReceive(_logger);
 
             var request = new HttpRequestMessage(HttpMethod.Get, url);
             SendUtils.PrepareHttpRequest(request, _httpOptions);
             request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/event-stream"));
             var response = await _httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+            if (!response.IsSuccessStatusCode)
```

Comment:
Why not use `response.EnsureSuccessStatusCode()` like the long polling transport does?

Reply:
You want the exception that is thrown by `EnsureSuccessStatusCode` to manifest on the TCS.

Having said that, I actually think `EnsureSuccessStatusCode` coupled with a `try`-`catch` that puts the exception into the TCS (like in Long Polling) would be good as it would capture the stack properly.

Reply:
Also, we need the `try`-`catch` anyway since not all errors will manifest as a failed response, some will manifest as exceptions thrown by `SendAsync` (socket errors such as not being able to reach the server will manifest that way)